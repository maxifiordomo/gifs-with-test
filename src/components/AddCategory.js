import React, { useState } from 'react'
import PropTypes from 'prop-types'



export const AddCategory = ({ setCategories }) => {

    const [inputValue, setinputValue] = useState('');
    //si yo dejo el useState en undefined (es decir vacio literalmente), me va a tirar un error.
    const handleInputChange = (e) => {
        setinputValue(e.target.value)
        console.log('Handleinputchange llamado');
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        //validacion
        console.log('Hola mundo', inputValue)
        if (inputValue.trim().length > 2) {
            //esto se trata de un callback, cats hace referencia a categories y por eso puede desestructurar el array.
            setCategories(cats => [inputValue, ...cats]);
            setinputValue('');
        }
    }

    return (
        <form onSubmit={handleSubmit}>
            <p>{inputValue}</p>
            <input
                type="text"
                value={inputValue}
                onChange={handleInputChange}
            />
        </form>
    )
}

AddCategory.propType = {
    setCategories: PropTypes.func.isRequired
}