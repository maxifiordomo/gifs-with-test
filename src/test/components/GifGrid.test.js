import { GifGrid } from "../../components/GifGrid"
import React from 'react';
import { shallow } from "enzyme";
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock('../../hooks/useFetchGifs')

describe('Pruebas en el componente GifGrid', () => {

    const category = 'Hola XD';

    test('Debe de hacer match con el snapShot', () => {

        //mockReturnValue devuelve todo lo que este dentro de si
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper = shallow(<GifGrid category={category} />);
        expect(wrapper).toMatchSnapshot();

    })
    test('Debe de mostrar items cuando se cargan imagenes useFetchGifs', () => {
        //este gifs se crea para pasarle informacion parecida a la que se le manda directamente en el componente
        const gifs = [{
            id: 'abc',
            url: 'https://localhost//cualquier/cosa.jpg',
            title: 'cualquier cosa',
        }];

        //data recibe a gifs y loading se cambia de true a false ya que no se muestra mas el loading(debido a que ya data contiene info)
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });

        const wrapper = shallow(<GifGrid category={category} />);


        expect(wrapper).toMatchSnapshot();
        //Se encarga de revisar en todo el render si existe lo que el wrapper busca
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);

    })

})
