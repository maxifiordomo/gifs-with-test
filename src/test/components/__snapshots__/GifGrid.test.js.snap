// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente GifGrid Debe de hacer match con el snapShot 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__bounced"
  >
    Hola XD
  </h3>
  <p>
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Pruebas en el componente GifGrid Debe de mostrar items cuando se cargan imagenes useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__bounced"
  >
    Hola XD
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="abc"
      key="abc"
      title="cualquier cosa"
      url="https://localhost//cualquier/cosa.jpg"
    />
  </div>
</Fragment>
`;
